File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/discover/model/DiscoverMovieResponse.java
package dev.reuss.tmdb.domain.discover.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.page.PagedResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Paginated TMDB movie discover response.
 *
 * @param page         current result page
 * @param results      discovered movies
 * @param totalPages   total available pages
 * @param totalResults total available results
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record DiscoverMovieResponse(
        int page,

        List<DiscoverMovie> results,

        @JsonProperty("total_pages")
        int totalPages,

        @JsonProperty("total_results")
        int totalResults
) implements PagedResponse<DiscoverMovie>, TmdbModel {

    public DiscoverMovieResponse {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/discover/model/DiscoverMovieResponse.kt
package dev.reuss.tmdb.domain.discover.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.page.PagedResponse

/**
 * Paginated TMDB movie discover response.
 *
 * @property page current result page
 * @property results discovered movies
 * @property totalPages total available pages
 * @property totalResults total available results
 */
@JvmRecord
data class DiscoverMovieResponse(
    @all:JsonProperty("page")
    override val page: Int,

    @all:JsonProperty("results")
    override val results: List<DiscoverMovie> = emptyList(),

    @all:JsonProperty("total_pages")
    override val totalPages: Int,

    @all:JsonProperty("total_results")
    override val totalResults: Int
) : PagedResponse<DiscoverMovie>, TmdbModel